### PR TITLE
feat: enable `on` key in `join_asof`

### PR DESCRIPTION
**Pull request by raisadz**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #896
- Closes #

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/dataframe.py [hunk 1]
```diff
@@ -3576,9 +3594,18 @@ def join_asof(
...
+            raise ValueError(msg)
+        if left_on is not None and right_on is not None:
+            return super().join_asof(
+                other, left_on=left_on, right_on=right_on, strategy=strategy
+            )
+        elif on is not None:
+            return super().join_asof(other, on=on, strategy=strategy)
+        else:
+            msg = "Either (`left_on` and `right_on`) or `on` keys should be specified."
+            raise ValueError(msg)
```
Comment (MarcoGorelli) [inline narwhals/dataframe.py:3608]: does this need updating, or can you also just defer to `super().join_asof(` here?

**Review by MarcoGorelli — COMMENTED** [review]
thanks @raisadz , looks great! just one comment

File: narwhals/dataframe.py (hunk 1 above)
Comment (raisadz) [reply narwhals/dataframe.py]: thanks, updated this part

**Review by MarcoGorelli — APPROVED** [review]
thanks @raisadz !

**Pull request merged**